client.go

```diff
@@ -10,15 +10,20 @@ import (
 
 //Client is a SSE client
 type Client struct {
-	r io.ReadCloser
+	r io.Reader
 	s *bufio.Scanner
 }
 
+// NewClient returns a client which will parse Event from the io.Reader
+func NewClient(r io.Reader) *Client {
```

Comment:
we should use this method in Connect

Reply:
I think `Connect` shouldn't be in this package, so it stays independent of the `http` package (if I get the event via some other method for instance).

Reply:
But in the meantime, I can move `Connect` to another file (and make it use `NewClient`)

Reply:
I dont get what you mean? Why should an HTTP protocol be entirely seperate from http?

Reply:
You are right, it doesn't make much sense.
For testing purposes, it might be easier to make a scanner and a client (the scanner being the underlying part, parsing the stream and the client handling the http connection)